### DISCUSSION TITLE
How are xcoms managed ? #newbie-questions

### DISCUSSION THREAD

**Discussion by Frqei** [apache/airflow#16222] [Q&A] [answered]
I recently switched to Airflow2 and try to learn the taskflow api.
Thus I'm now using return values which are stored in xcoms that I didn't used before.
I would like to ask how are xcoms managed ?
Will they be deleted automatically ? because the number of xcoms listed in the dedicated tab is growing even if dags ended succesfully.
Regards
Frqei

**Comment by potiuk** [comment, marked as answer]
You can read more here: https://airflow.apache.org/docs/apache-airflow/stable/concepts/xcoms.html - default xcom backend is metadata db but in 2.0 you can add custom backends . In DB there is no built in retention (there are 3rd party maintenance DAGs that are often used for retention. 

https://github.com/teamclairvoyant/airflow-maintenance-dags )

You can also see this nice article from Astronomer about custom xcom backends 
https://www.astronomer.io/guides/custom-xcom-backends

**Reply by Frqei** [reply]
Thank you very much, I saw the documentation but didn't find it explicit at first glance.
With airflow 1 I used redis as cache to pass data between tasks.
To stay consistent with the taskflow api I'll go for a redis as xcom backend instead of just setting and getting keys in tasks.
Regards
Frqei
